### PR TITLE
Improve Fuse.statfs to avoid copyProperties and cache on client side

### LINKED ISSUE #15295: Fuse.statfs impact Alluxio performance and cause OOM
**Alluxio Version:**
2.8.0

**Describe the bug**
Fuse.statfs called `ClientContext.create` for each call.
```
  private int statfsInternal(String path, Statvfs stbuf) {
    ClientContext ctx = ClientContext.create(mConf);
```
Each `ClientContext.create` will 
```
mClusterConf = new InstancedConfiguration(alluxioConf.copyProperties(),
          alluxioConf.clusterDefaultsLoaded());
```
copy the whole alluxio properties.

So each `Fuse.statfs` copies the whole property key.
Kernel calls `Fuse.statfs` a large number of times and will likely cause memory/GC overhead, and may cause OOM.

Each `Fuse.statfs` will issue one master RPC call to master, the current implementation adds a lot of master RPCs. A memorization with expiration will help resolve the issue. 
**To Reproduce**
Steps to reproduce the behavior (as minimally and precisely as possible)

**Expected behavior**

**Urgency**
Describe the impact and urgency of the bug.

**Are you planning to fix it**
Please indicate if you are already working on a PR. 

**Additional context**

### PR DESCRIPTION
**Pull request by LuQQiu**
### What changes are proposed in this pull request?

Fixes #15295

### Why are the changes needed?

Cache the capacity info for Fuse.statfs on Fuse side to reduce the master RPC calls.
Improve the Fuse.statfs() to avoid create client context and copyProperties for each call.

### Does this PR introduce any user facing changes?
NA

### CODE REVIEW

**Comment by LuQQiu** [comment]
Error: 1.631 [ERROR] alluxio.client.cli.JournalToolTest.dumpSimpleEmbeddedJournal
Error: 1.631 [ERROR]   Run 1: JournalToolTest.dumpSimpleEmbeddedJournal » Runtime java.lang.RuntimeException...
Error: 1.631 [ERROR]   Run 2: JournalToolTest.dumpSimpleEmbeddedJournal » Runtime java.lang.RuntimeException...
18:33:21.631 [INFO] 

Error: 6.695 [ERROR] alluxio.client.cli.JournalToolTest.dumpSimpleEmbeddedJournal  Time elapsed: 60.388 s  <<< ERROR!
java.lang.RuntimeException: java.lang.RuntimeException: Timed out waiting for process Alluxio worker @1e4947aa886c:0 to start
	at alluxio.testutils.LocalAlluxioClusterResource$1.evaluate(LocalAlluxioClusterResource.java:199)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:288)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:282)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.RuntimeException: Timed out waiting for process Alluxio worker @1e4947aa886c:0 to start
	at alluxio.master.TestUtils.waitForReady(TestUtils.java:34)
	at alluxio.master.AbstractLocalAlluxioCluster.startWorkers(AbstractLocalAlluxioCluster.java:160)
	at alluxio.master.AbstractLocalAlluxioCluster.start(AbstractLocalAlluxioCluster.java:88)
	at alluxio.testutils.LocalAlluxioClusterResource.start(LocalAlluxioClusterResource.java:161)
	at alluxio.testutils.LocalAlluxioClusterResource$1.evaluate(LocalAlluxioClusterResource.java:196)
	... 5 more

**Comment by LuQQiu** [comment]
@Xenorith PR for 2.8, FYI

File: tests/src/test/java/alluxio/client/fuse/AbstractFuseIntegrationTest.java [hunk 1]
```diff
@@ -109,8 +113,10 @@ public void before() throws Exception {
     IntegrationTestUtils.reserveMasterPorts();
     ServerConfiguration.global().validate();
     mAlluxioCluster.start();
-    mFileSystem = mAlluxioCluster.getClient();
-    mountFuse(mFileSystem, mMountPoint, ALLUXIO_ROOT);
+    // TODO(lu) how to get FileSystemContext
```
Comment (Xenorith) [inline tests/src/test/java/alluxio/client/fuse/AbstractFuseIntegrationTest.java:116]: TODONE?

**Review by ssz1997 — APPROVED** [review]
LGTM in general. It does seem weird to me that a field of my filed is my field. In this case, the fsContext is a field of both AlluxioFuseFileSystem and fs, a field of AlluxioFuseFileSystem.

**Comment by ssz1997** [comment]
> > LGTM in general. It does seem weird to me that a field of my filed is my field. In this case, the fsContext is a field of both AlluxioFuseFileSystem and fs, a field of AlluxioFuseFileSystem.
> 
> FileSystem doesn't have method of getContext to get the FileSystemContext which i think also make sense. Since the place that we allow to use FileSystem operations we may not want they to be able to get FileSystemContext and do all the random operations with the clients

I agree we shouldn't have a getConext(), totally unnecessary and might be harmful. Yeah I just have the feeling but no suggestions

**Comment by LuQQiu** [comment]
alluxio-bot, merge this please

**Comment by LuQQiu** [comment]
@Xenorith can you help cherry-pick this change to 2.8?

**Comment by Xenorith** [comment]
alluxio-bot, cherry-pick this to branch-2.8 please

**Pull request merged**